routers/web/user/setting/account.go

```diff
@@ -246,6 +246,13 @@ func DeleteAccount(ctx *context.Context) {
 		return
 	}
 
+	// admin should not delete themself
+	if ctx.Doer.IsAdmin {
+		ctx.Flash.Error(ctx.Tr("admin.users.cannot_delete_self"))
```

Comment:
Just a small suggestion.

`admin.users.cannot_delete_self` ("You cannot delete yourself") is used to inform the user that they cannot delete themselves on the admin page. However, this is the account page, and the user is indeed deleting themselves. So, maybe the error message could be something like "You cannot delete yourself when you are an admin. Please remove your admin privileges first."